packages/core/src/exec.ts

```diff
@@ -0,0 +1,212 @@
+
+        if (typeof sendAction.params.delay === 'number') {
+          interpreter.defer(sendAction);
+          return;
+        } else {
+          if (sendAction.params.to) {
+            const target = sendAction.params.to;
+            execSendTo(sendAction.params._event, target, actorCtx);
+          } else {
+            interpreter.send(sendAction.params._event as SCXML.Event<any>);
+          }
+        }
+      },
+      [actionTypes.cancel]: (_ctx, _e, { action }) => {
+        interpreter.cancel((action as CancelActionObject).params.sendId);
+      },
+      [actionTypes.invoke]: (_ctx, _e, { action }) => {
+        const { id, autoForward, ref } = (action as InvokeActionObject).params;
+        if (!ref) {
+          if (!IS_PRODUCTION) {
+            warn(
+              false,
+              `Actor type '${
+                ((action as InvokeActionObject).params
+                  .src as InvokeSourceDefinition).type
+              }' not found in machine '${interpreter.id}'.`
+            );
+          }
+          return;
+        }
+        ref._parent = interpreter; // TODO: fix
+        // If the actor didn't end up being in the state
+        // (eg. going through transient states could stop it) don't bother starting the actor.
+        if (!state.children[id]) {
+          state.children[id] = ref;
+        }
+        actorCtx.defer?.(() => {
+          try {
+            if (autoForward) {
+              interpreter._forwardTo.add(id);
+            }
+
+            ref.start?.();
+          } catch (err) {
+            interpreter.send(error(id, err));
+            return;
+          }
+        });
+      },
+      [actionTypes.stop]: (_ctx, _e, { action }) => {
+        const { actor } = (action as StopActionObject).params;
+
+        if (actor) {
+          actor.stop?.();
+        }
+      },
+      [actionTypes.log]: (_ctx, _e, { action }) => {
+        const { label, value } = (action as LogActionObject).params;
+
+        if (label) {
+          actorCtx.logger?.(label, value);
+        } else {
+          actorCtx.logger?.(value);
+        }
+      }
+    } as ActionFunctionMap<ContextFrom<TState>, EventFrom<TState>>)[actionType]
+  );
+}
+
+function execSendTo(
+  event: SCXML.Event<AnyEventObject>,
+  to: ActorRef<any>,
+  actorContext: ActorContext<any, any>
+) {
+  const interpreter = actorContext.self;
+  const target = to;
+
+  if (!target) {
+    const executionError = new Error(
+      `Unable to send event to child '${to}' from service '${interpreter.name}'.`
+    );
+    interpreter.send(
+      toSCXMLEvent<any>(actionTypes.errorExecution, {
+        data: executionError as any // TODO: refine
+      }) as any // TODO: fix
+    );
+
+    // tslint:disable-next-line:no-console
+    if (!IS_PRODUCTION) {
+      warn(
+        false,
+        `Service '${interpreter.name}' has no parent: unable to send event ${event.type}`
+      );
+    }
+    return;
+  }
```

Comment:
This whole thing is a code that is never being called. The only call to this `execSendTo` is here:
https://github.com/statelyai/xstate/blob/907288c688c6179db2ed09efd66e5ad0d5ebbd21/packages/core/src/exec.ts#L110-L112
and the `target`/`to` is first checked before calling this. This is also reflected in the types here because `to`/`target` is of type `ActorRef<any>` and it gets narrowed to `never` in this `if` block